File: rust/src/model/domain/vendor.rs
#[derive(Debug, Clone, Copy)]
pub struct Vendor(pub u32);

impl Vendor {
    pub fn id(&self) -> String {
        format!("VEND-{:05}", self.0)
    }

    pub fn name(&self) -> String {
        format!("Vendor {:05}", self.0)
    }
}

File: golang/model/domain/vendor.go
package domain

import (
	"fmt"
)

type Vendor struct {
	no int
}

func NewVendor(no int) Vendor {
	return Vendor{no}
}

func (u Vendor) Id() string {
	return fmt.Sprintf("VEND-%05d", u.no)
}

func (u Vendor) Name() string {
	return fmt.Sprintf("Vendor %05d", u.no)
}

func (u Vendor) No() int {
	return u.no
}
